src/scheduler/fifo.c

```diff
@@ -2327,3 +2350,357 @@ next_job(status *policy, server_info *sinfo, int flag)
+				if (copy_attr_value(&tmp_partitions, attr->value) == 0)
+					return 0;
+			} else if (!strcmp(attr->name, ATTR_comment)) {
+				if (copy_attr_value(&tmp_comment, attr->value) == 0)
+					return 0;
+			}
+		}
+		attr = attr->next;
+	}
+
+	if (!dflt_sched) {
+		int err;
+		int priv_dir_update_fail = 0;
+		struct attropl *patt;
+		char comment[MAX_LOG_SIZE] = {0};
+		static char *priv_dir = NULL;
+		if ((log_dir != NULL) && strcmp(log_dir, tmp_log_dir) != 0) {
+			(void)snprintf(path_log,  sizeof(path_log), tmp_log_dir);
+			log_close(1);
+			if (log_open(logfile, path_log) == -1) {
+				/* update the sched comment attribute with the reason for failure */
+				attribs = calloc(2, sizeof(struct attropl));
+				if (attribs == NULL) {
+					snprintf(log_buffer, sizeof(log_buffer), "can't update scheduler attribs, calloc failed");
+					log_err(errno, __func__, log_buffer);
+					free(tmp_log_dir);
+					free(tmp_priv_dir);
+					free(tmp_partitions);
+				}
+
+				strncpy(comment, "Unable to change the sched_logs directory", MAX_LOG_SIZE -1);
+				patt = attribs;
+				patt->name = ATTR_comment;
+				patt->value = comment;
+				patt->next = patt + 1;
+				patt++;
+				patt->name = ATTR_scheduling;
+				patt->value = "0";
+				patt->next = NULL;
+
+				err = pbs_manager(connector,
+					MGR_CMD_SET, MGR_OBJ_SCHED,
+					sc_name, attribs, NULL);
+				free(attribs);
+				if (err) {
+					snprintf(log_buffer, sizeof(log_buffer), "Failed to update scheduler comment %s at the server", comment);
+					log_err(-1, __func__, log_buffer);
+				}
+				log_dir = tmp_log_dir;
+				return 0;
+			} else {
+				if (tmp_comment != NULL)
+					clear_comment = 1;
+				snprintf(log_buffer, sizeof(log_buffer), "scheduler log directory is changed to %s", tmp_log_dir);
+				schdlog(PBSEVENT_SCHED, PBS_EVENTCLASS_SCHED, LOG_INFO,
+						"reconfigure", log_buffer);
+				free(log_dir);
+			}
+		}
+		log_dir = tmp_log_dir;
+
+		if ((priv_dir != NULL) && strcmp(priv_dir, tmp_priv_dir) != 0) {
+			int c;
+			#if !defined(DEBUG) && !defined(NO_SECURITY_CHECK)
+				c  = chk_file_sec(tmp_priv_dir, 1, 0, S_IWGRP|S_IWOTH, 1);
+				c |= chk_file_sec(pbs_conf.pbs_environment, 0, 0, S_IWGRP|S_IWOTH, 0);
+				if (c != 0) {
+					snprintf(log_buffer, sizeof(log_buffer), "PBS failed validation checks for directory %s", tmp_priv_dir);
+					log_err(-1, __func__, log_buffer);
+					strncpy(comment, "PBS failed validation checks for sched_priv directory", MAX_LOG_SIZE -1);
+					priv_dir_update_fail = 1;
+				}
+			#endif  /* not DEBUG and not NO_SECURITY_CHECK */
+			if (c == 0) {
+				if (chdir(tmp_priv_dir) == -1) {
+					snprintf(log_buffer, sizeof(log_buffer), "PBS failed validation checks for directory %s", tmp_priv_dir);
+					strncpy(comment, "PBS failed validation checks for sched_priv directory", MAX_LOG_SIZE -1);
+					log_err(-1, __func__, log_buffer);
+					priv_dir_update_fail = 1;
+				} else {
+					int lockfds;
+					(void)unlink("sched.lock");
```

Comment:
Two schedulers can not have the same priv directory. This is not possible since we bail out at the early stage itself even if the admin tries to give the same directory for multiple sched objects.